app/models/work_completion/tube_rack_completion.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+# Class WorkCompletion::TubeRackCompletion provides the business logic
+# for passing tubes within a tube rack, especially in the Limber pipelines. This has
+# been pulled out of WorkCompletion itself to allow for separate behaviour
+# for tube racks.
+#
+# @author Genome Research Ltd.
+#
+class WorkCompletion::TubeRackCompletion < WorkCompletion::LabwareCompletion
+  # Connects upstream requests to the target labware's racked tubes.
+  #
+  # Iterates through each upstream request and links it to the receptacle
+  # of each tube in the target labware's racked tubes.
+  #
+  # @return [void]
+  #
+  # @example
+  #   connect_requests
+  def connect_requests
+    detect_upstream_requests.each do |upstream|
+      target_labware.racked_tubes.each { |tube| pass_and_link_up_requests(tube.tube.receptacle, upstream) }
+    end
+    @order_ids.uniq!
```

Comment:
what is the order ids uniq line doing? (add comment?)
if you are expecting all tubes to have the same submission that may not be true in the same way that is was for wells on a plate